projects/npm-tools/packages/npm-scripts/src/sass/build.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const fs = require('fs');
+const globby = require('globby');
```

Comment:
👍  I didn't delve to deep into this, I just noticed it as an already existing dependency and went with it.

Reply:
@wincent I tried using `getPaths` but it seems to require that the directory to glob from always starts at `'.'`[here](https://github.com/liferay/liferay-frontend-projects/blob/master/projects/npm-tools/packages/npm-scripts/src/utils/expandGlobs.js#L206).

For processing sass, I think we need to be able to specify the absolute path of where the scss is located. I could be missing something here, but I think we need to update `getPaths` and `expandGlobs` to accommodate for a different base directory and also needs to support a list of excludes rather than a file path for excluded files.

I can go through and make these changes, but I didn't want to mess with those utils quite yet in case I broke something that was dependent on the starting directory for traverse.

Reply:
> For processing sass, I think we need to be able to specify the absolute path of where the scss is located. I could be missing something here, but I think we need to update `getPaths` and `expandGlobs` to accommodate for a different base directory and also needs to support a list of excludes rather than a file path for excluded files.
>
> I can go through and make these changes, but I didn't want to mess with those utils quite yet in case I broke something that was dependent on the starting directory for traverse.

I guess we can make the change to not traverse from `'.'` _if we are sure that we need to_ — that was just a simplifying assumption made in the name of robustness (because if you want to support building "project x" from some other directory belonging "project y", then you suddenly have 100s of places in the code base where you have to uphold the convention rather than just assuming that `process.cwd()` is always set to "the right place"... which sounds like a recipe for bugs). What's the reason we need absolute paths here again?

As for:

> also needs to support a list of excludes rather than a file path for excluded files

It already supports [negated globs](https://github.com/liferay/liferay-frontend-projects/blob/542e42c5b24897a1e63016afae7c44e7351e526e/projects/npm-tools/packages/npm-scripts/src/utils/getRegExpForGlob.js#L63), so maybe you don't need that?

Reply:
> What's the reason we need absolute paths here again?

The funny thing is, we might not need to in portal to reference anything outside of `'.'` in portal, but when it comes to our unit tests, we have to move back to the root of this mono-repo to reference the `__fixtures__` directory.


> It already supports negated globs, so maybe you don't need that?

👍  didn't realize that